packages/date-picker/test/dialog.test.js

```diff
@@ -26,30 +25,71 @@ describe('modeless dialog', () => {
     await nextFrame();
   });
 
-  function touchstart(node) {
-    const event = new CustomEvent('touchstart', { bubbles: true, composed: true });
-    const nodeRect = node.getBoundingClientRect();
-    const clientX = nodeRect.left;
-    const clientY = nodeRect.top;
-    event.touches = event.changedTouches = event.targetTouches = [{ clientX, clientY }];
-    node.dispatchEvent(event);
-  }
-
-  it('should not end up behind the dialog overlay on mousedown', async () => {
-    datepicker.opened = true;
-    datepicker.dispatchEvent(new CustomEvent('mousedown', { bubbles: true, composed: true }));
-    await nextFrame();
-    expect(parseFloat(getComputedStyle(datepicker.$.overlay).zIndex)).to.equal(
-      parseFloat(getComputedStyle(dialog.$.overlay).zIndex) + 1,
-    );
+  describe('modal', () => {
+    beforeEach(async () => {
+      datepicker.inputElement.focus();
+      await open(datepicker);
+      await nextRender();
+    });
+
+    it('should not close the dialog when closing date-picker on input element Escape', async () => {
+      await sendKeys({ press: 'Escape' });
+
+      expect(datepicker.opened).to.be.false;
+      expect(dialog.opened).to.be.true;
+    });
+
+    it('should not close the dialog when closing date-picker on month calendar Escape', async () => {
+      // Focus the month calendar
+      await sendKeys({ press: 'Tab' });
+
+      await sendKeys({ press: 'Escape' });
+
+      expect(datepicker.opened).to.be.false;
+      expect(dialog.opened).to.be.true;
+    });
+
+    it('should not close the dialog when closing date-picker on Today button Escape', async () => {
+      // Focus the Today button
+      getOverlayContent(datepicker).$.todayButton.focus();
+
+      await sendKeys({ press: 'Escape' });
+
+      expect(datepicker.opened).to.be.false;
+      expect(dialog.opened).to.be.true;
+    });
+
+    it('should not close the dialog when closing date-picker on Today button Escape', async () => {
```

Comment:
We could also add a test that verifies that pressing escape twice actually closes the dialog.

Reply:
Thanks, done.